### ISSUE TITLE
[Feature Request] Delete entries in autocorrection

### ISSUE THREAD

**Issue by TheEvilSkeleton** [dslul/openboard#313]
Gboard has a neat little feature to manually remove unwanted entries in the autocorrection field, by holding down an entry and dragging it to the garbage icon.

I'd like to have such feature, since it sometimes autocorrects to unwanted words everytime I write a specific word.

**Comment by Candlemass** [comment]
So, any plans to offer this and bring openboard to feature parity with other third party offerings?

**Comment by MajeurAndroid** [comment]
As always, it can be added to the todo list, but is not planned for now.
Feel free to try implent this on your own, but it might be a tricky one.

**Comment by Mrnofish** [comment]
Drag and drop seems nice but perhaps overkill?

@MajeurAndroid There's a menu that opens when tapping a word, would it be a lot of work to add a menu entry to purge that suggestion?

It would be a massive QoL improvement IMO.

**Comment by Candlemass** [comment]
> Drag and drop seems nice but perhaps overkill?
> 
> @MajeurAndroid There's a menu that opens when tapping a word, would it be a lot of work to add a menu entry to purge that suggestion?
> 
> It would be a massive QoL improvement IMO.

Apparently it takes many YEARS to implement basic bone standard features that every single other keyboard basically offered at day one. Instead we wait and continually be annoyed at terrible suggestions that get recommended over and over again that we'll never use and have no way to get rid of. Fantastic.

**Comment by AdamPS** [comment]
OpenBoard is already a great keyboard. It has features that even GBoard doesn't, for example the menu that opens on tapping a completion.

> add a menu entry to purge that suggestion?

I agree that would be a good interface. Drag/drop wouldn't fit so well given the menu exists.

**Comment by Mrnofish** [comment]
On second thought I'm not sure that's possible because that menu may be part of the spellchecker, which I assume is provided by the OS. (The keyboard likely only feeds the suggestions themselves to the spellchecker)

And that would explain why GBoard uses the drag'n'drop mechanism.

Anyway, no work being done on OpenBoard so... ¯\\\_(ツ)_/¯

**Comment by Helium314** [comment]
> On second thought I'm not sure that's possible because that menu may be part of the spellchecker, which I assume is provided by the OS. (The keyboard likely only feeds the suggestions themselves to the spellchecker)
> 
> And that would explain why GBoard uses the drag'n'drop mechanism.

I think you're right. A while ago I played with this and concluded that the menu is not part of the keyboard.
For myself I replaced the "more suggestions" long press menu with an option to remove the suggestion (put it in some sort of blacklist).
But no reason to bother with a PR...

**Comment by Candlemass** [comment]
> OpenBoard is already a great keyboard. It has features that even GBoard doesn't, for example the menu that opens on tapping a completion.
> 

Of which is irrelevant when basic features that severely hinder the user experience are this horribly broken and we get nothing but crickets from the devs.

**Comment by AdamPS** [comment]
@Candlemass Of course we will each of us have our own point of view, and I respect yours.

Personally I see it like this: OpenBoard is the best keyboard I've found that respects my privacy and it has almost all the features I want. The inconveniences are fairly minor, and I tried several other keyboard that were much worse. I'm hugely grateful that the developers have put in a lot of work to create it, then amazingly they share it absolutely for free, expecting nothing in return.

**Comment by Candlemass** [comment]
@AdamPS

You speak as if this is exclusive to this project when really it's just a tiny branch of the tree that is AOSP. When the devs clearly ignore a critical problem that has been ongoing for years and reported multiple times they start losing merit and the benefit of the doubt when it comes to what the intended purpose of the project is/how committed they are to actually offering a coherent user experience. Regarding your other keyboard comparisons that's entirely subjective. My points and references are factual. There's a big difference.

**Comment by AdamPS** [comment]
@Candlemass If you are criticising the Google AOSP [lack of] development then for sure I can agree with that. If you are criticising the devs of this project (who I believe are giving their time for free), then I would encourage you to ask yourself why you do it. I'm pretty sure you won't encourage them to do what you want. If you strongly disagree with my feeling that this is the best keyboard then why use it, or participate in this community? We can each of us like and use different software without any need to spread hostility towards the ones we choose not to use.

**Comment by Candlemass** [comment]
@AdamPS

I'm critisizing both because there are obvious deficiencies not only in lack of addrsssing problems that have existed for YEARS but essentially ignoring this issue for example and hoping it will some how magically fix itself. Plus I have every right to do so and I'm exercising that freedom. Silence only acts to enable and support such poor antics.

**Comment by Mrnofish** [comment]
While wholeheartedly I agree with you, @AdamPS, on not abusing people (something that tons of rude devs with shitty attitudes should keep in mind, too), there is an implied contract between devs and the community, and while nobody debates the dev team have been kindly donating their resources to further a project that benefits the general Android community, not everything can be glossed over with simply a *"but hey it's free"*.

Between OpenBoard being (IMHO) the best FOSS keyboard available ATM, and promises that were made, understandably there is no small amount of frustration and disappointment among those who use(d) it as a daily driver.

This sentiment is compounded because FOSS keyboards as a whole seem to be treading water ATM, and that leaves users with a thorny and painful dilemma, choosing between more functional but non-FOSS keyboards (with the associated concerns), or a subpar experience.

It is further exacerbated, for some of the issues that make OpenBoard frustrating to use, could have been fixed or at least alleviated by releasing new builds that incorporate ready-made PRs that are simply collecting dust.

e.g. #576 #749 #750 #618 and I remember one change that tuned the suggestion strength, addressing my biggest grievance with OpenBoard since it entails the keyboard constantly prioritizing mistakes over corrections, after a long enough use it does leave a pretty sore spot behind, as I can attest.

**Comment by AdamPS** [comment]
@Mrnofish Thank you for a very clear, well-reasoned and polite comment.

I am an open-source developer, but not for Android. My own feeling is that I give my time for free and I'm free to choose how to spend it. This gives me a totally different feeling compared with paid programming work. Other people are then free to use the software, or not. They are also free to make suggestions, which I am free to follow or ignore. Of course if I ignore all of the suggestions then I will have few users, but that's my choice, it's my free time. The software is free from any cost, and free from any warrantee or support contract. The original developers can lose enthusiasm and move away, however the amazing thing about FOSS is that anyone else is completely free to step in and contribute. People that are frustrated are free to volunteer to be maintainers. If the original devs have completely gone away ignoring requests, then people are free to fork the project.

In summary, the "free" in FOSS includes many freedoms: free from cost; free to use as you like; free to modify; free to inspect the software that you use and importantly, freedom for the developers to spend their free time exactly how they like.

So I don't really agree there is an implied "contract". In any case, what have the community given? If people would donate as much as a paid ad-free app would cost, FOSS would be thriving. Sadly in my experience, donations tend to be few and far between, maybe 1% of the commercial rate. Combined with the amount of criticism that can be faced in the issue queue, sadly it is indeed leading to a decline in FOSS software globally.

**Comment by TheEvilSkeleton** [comment]
Quick reminder that this issue is about deleting entries in the correction suggestions field, not about the philosophies of FOSS and what developers and maintainers should prioritize.

**Comment by Candlemass** [comment]
LMAO at this STILL not being addressed....

**Comment by TheEvilSkeleton** [comment]
@Candlemass https://boyter.org/posts/the-three-f-s-of-open-source/

**Comment by Dellum722** [comment]
> @Candlemass https://boyter.org/posts/the-three-f-s-of-open-source/

@TheEvilSkeleton

He has a point though.

Rather than link to some ideological nonsense written by some bitter individual that likely failed to live up to the expectations they set for themselves, how about you prove him wrong?
